flow/Platform.actor.cpp

```diff
@@ -2521,69 +2460,14 @@ StringRef separator(&separatorChar, 1);
 StringRef dotdot = ".."_sr;
 
 std::string cleanPath(std::string const& path) {
-	std::vector<StringRef> finalParts;
-	bool absolute = !path.empty() && path[0] == CANONICAL_PATH_SEPARATOR;
-
-	StringRef p(path);
-
-	while (p.size() != 0) {
-		StringRef part = p.eat(separator);
-		if (part.size() == 0 || (part.size() == 1 && part[0] == '.'))
-			continue;
-		if (part == dotdot) {
-			if (!finalParts.empty() && finalParts.back() != dotdot) {
-				finalParts.pop_back();
-				continue;
-			}
-			if (absolute) {
-				continue;
-			}
-		}
-		finalParts.push_back(part);
-	}
-
-	std::string result;
-	result.reserve(PATH_MAX);
-	if (absolute) {
-		result.append(1, CANONICAL_PATH_SEPARATOR);
-	}
-
-	for (int i = 0; i < finalParts.size(); ++i) {
-		if (i != 0) {
-			result.append(1, CANONICAL_PATH_SEPARATOR);
-		}
-		result.append((const char*)finalParts[i].begin(), finalParts[i].size());
-	}
-
-	return result.empty() ? "." : result;
+	std::filesystem::path p = std::filesystem::path(path);
+	p = std::filesystem::absolute(p);
+	p = std::filesystem::canonical(p);
+	return p.string();
 }
 
 std::string popPath(const std::string& path) {
-	int i = path.size() - 1;
-	// Skip over any trailing separators
-	while (i >= 0 && path[i] == CANONICAL_PATH_SEPARATOR) {
-		--i;
-	}
-	// Skip over non separators
-	while (i >= 0 && path[i] != CANONICAL_PATH_SEPARATOR) {
-		--i;
-	}
-	// Skip over trailing separators again
-	bool foundSeparator = false;
-	while (i >= 0 && path[i] == CANONICAL_PATH_SEPARATOR) {
-		--i;
-		foundSeparator = true;
-	}
-
-	if (foundSeparator) {
-		++i;
-	} else {
-		// If absolute then we popped off the only path component so return "/"
-		if (!path.empty() && path.front() == CANONICAL_PATH_SEPARATOR) {
-			return "/";
-		}
-	}
-	return path.substr(0, i + 1);
+	return (std::filesystem::path(path).parent_path().filename()).string() + "/";
```

Comment:
kind of violate the law of demetra

Reply:
I think this is fixed in my latest commit but feel free to correct me.